File: lesson_8/app/src/main/java/com/kalashnyk/denys/gmapsapp/di/module/RepositoryModule.java
package com.kalashnyk.denys.gmapsapp.di.module;

import com.kalashnyk.denys.gmapsapp.di.scope.RepositoryScope;
import com.kalashnyk.denys.gmapsapp.repository.AppRepository;
import com.kalashnyk.denys.gmapsapp.repository.database.dao.PinDao;
import com.kalashnyk.denys.gmapsapp.repository.server.ServerCommunicator;

import dagger.Module;
import dagger.Provides;

@Module
public class RepositoryModule {
    @RepositoryScope
    @Provides
    AppRepository providesRepository(ServerCommunicator communicator, PinDao productDAO) {
        return new AppRepository(communicator, productDAO);
    }
}

File: lesson_9/app/src/main/java/com/kalashnyk/denys/kotlinbindingsample/di/module/RepositoryModule.kt
package com.kalashnyk.denys.kotlinbindingsample.di.module

import com.kalashnyk.denys.kotlinbindingsample.di.scope.RepositoryScope
import com.kalashnyk.denys.kotlinbindingsample.repository.AppRepository
import com.kalashnyk.denys.kotlinbindingsample.repository.database.AppDatabase
import com.kalashnyk.denys.kotlinbindingsample.repository.server.ServerCommunicator
import dagger.Module
import dagger.Provides

@Module
class RepositoryModule {
    @RepositoryScope
    @Provides
    internal fun providesRepository(communicator: ServerCommunicator, database: AppDatabase): AppRepository {
        return AppRepository(communicator, database)
    }
}